src/wallet.js

```diff
@@ -0,0 +1,273 @@
+   * @param {number} addressIndex - the index of the address
+   * @return {String} address
+   */
+  getBitcoinAddress(addressIndex: number): string {
+    return this.getBitcoinNode(addressIndex).getAddress()
+  }
+
+  /**
+   * Get the private key hex-string for a given bitcoin receive address
+   * @param {number} addressIndex - the index of the address
+   * @return {String} the hex-string. this will be either 64
+   * characters long to denote an uncompressed bitcoin address, or 66
+   * characters long for a compressed bitcoin address.
+   */
+  getBitcoinPrivateKey(addressIndex: number): string {
+    return getNodePrivateKey(this.getBitcoinNode(addressIndex))
+  }
+
+  /**
+   * Get the root node for the bitcoin public keychain
+   * @return {String} base58-encoding of the public node
+   */
+  getBitcoinPublicKeychain(): string {
+    return this.getBitcoinPrivateKeychain().neutered().toBase58()
+  }
+
+  /**
+   * Get the root node for the identity public keychain
+   * @return {String} base58-encoding of the public node
+   */
+  getIdentityPublicKeychain(): string {
+    return this.getIdentityPrivateKeychain().neutered().toBase58()
+  }
+
+  static getNodeFromBitcoinKeychain(keychainBase58: string, addressIndex: number,
+                                    chainType: string = EXTERNAL_ADDRESS): HDNode {
+    let chain
+    if (chainType === EXTERNAL_ADDRESS) {
+      chain = 0
+    } else if (chainType === CHANGE_ADDRESS) {
+      chain = 1
+    } else {
+      throw new Error('Invalid chain type')
+    }
+    const keychain = HDNode.fromBase58(keychainBase58)
+
+    return keychain.derive(chain).derive(addressIndex)
+  }
+
+  /**
+   * Get a bitcoin address given a base-58 encoded bitcoin node
+   * (usually called the account node)
+   * @param {String} keychainBase58 - base58-encoding of the node
+   * @param {number} addressIndex - index of the address to get
+   * @param {String} chainType - either 'EXTERNAL_ADDRESS' (for a
+   * "receive" address) or 'CHANGE_ADDRESS'
+   * @return {String} the address
+   */
+  static getAddressFromBitcoinKeychain(keychainBase58: string, addressIndex: number,
+                                       chainType: string = EXTERNAL_ADDRESS): string {
+    return BlockstackWallet
+      .getNodeFromBitcoinKeychain(keychainBase58, addressIndex, chainType)
+      .getAddress()
+  }
+
+  /**
+   * Get a ECDSA private key hex-string for an application-specific
+   *  address.
+   * @param {String} appsNodeKey - the base58-encoded private key for
+   * applications node (the `appsNodeKey` return in getIdentityKeyPair())
+   * @param {String} salt - a string, used to salt the
+   * application-specific addresses
+   * @param {String} appDomain - the appDomain to generate a key for
+   * @return {String} the private key hex-string. this will be a 64
+   * character string
+   */
+  static getLegacyAppPrivateKey(appsNodeKey: string, salt: string, appDomain: string): string {
```

Comment:
Same as above, should be instance method

Reply:
Same as above -- if this is an instance method, then you need to have the master key in memory to get this key.